core/includes/backup/backup.database.inc

```diff
@@ -0,0 +1,351 @@
+<?php
+/**
+ * @file
+ * Functions to handle the direct to/from database backup source.
+ */
+
+/**
+ * A destination type for saving to a database server.
+ *
+ * @ingroup backup_restore
+ */
+abstract class BackupDatabase extends Backup {
+  /**
+   * @var DatabaseConnection
+   *
+   * The database connection to be used when doing a backup or restore.
+   */
+  protected $connection = NULL;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function typeLabel() {
+    return t('Database');
+  }
+
+  /**
+   * Get the file type supported when backing-up or restoring to/from a file.
+   */
+  public function getFileTypeId() {
+    return 'sql';
+  }
+
+  /**
+   * Get the version info for the given database.
+   */
+  public function databaseInfo() {
+    return array(
+      'type' => FALSE,
+      'version' => t('Unknown'),
+    );
+  }
+
+  /**
+   * Get the default settings for this object.
+   *
+   * @return array
+   *   The default tables whose data can be ignored. These tables mostly
+   *   contain info which can be easily reproduced (such as cache or search
+   *   index) but also tables which can become quite bloated but are not
+   *   necessarily extremely important to back up or migrate during development
+   *   (such as access log and watchdog).
+   */
+  public function defaultSettings() {
+    $settings = parent::defaultSettings();
+
+    $all_tables = $this->getTableNames();
+
+    // Basic modules that should be excluded.
+    $basic = array(
+      // Default core tables.
+      'session',
```

Comment:
As we do not dump `session` table data, we might also want to not dump `batch` table data. I just realized, that the "backup" batch lands in the dump. :wink:

Reply:
I had excluded the session table in testing at one point, but I actually think it's good to include so that all users don't get logged out on restore.

Though interesting point about `batch`... maybe it should be excluded (nodata). I was also considering that maybe the `tempstore` table should be excluded (which holds form state and views/layout in-progress changes).